File: findMaxAndMin.java
public class findMaxAndMin {
    public static class Pair {
        long first;
        long second;

        public Pair(long first, long second) {
            this.first = first;
            this.second = second;
        }
    }

    public static Pair getMinMax(long a[], long n) {
        long ans1 = min(a, n);
        long ans2 = max(a, n);
        Pair p = new Pair(ans1, ans2);
        return p;
    }

    public static long min(long arr[], long n) {
        long min = Long.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            if (min > arr[i]) {
                min = arr[i];
            }
        }
        return min;
    }

    public static long max(long arr[], long n) {
        long max = Long.MIN_VALUE;
        for (int i = 0; i < n; i++) {
            if (max < arr[i]) {
                max = arr[i];
            }
        }
        return max;
    }




    // Don't write main function if you are coding in gfg
    public static void main(String[] args) {
        long[] arr = { 10, 5, 8, 2, 7 };
        int n = arr.length;

        Pair result = getMinMax(arr, n);

        System.out.println("Minimum: " + result.first);
        System.out.println("Maximum: " + result.second);
    }
}

File: findMaxAndMin.cpp
#include <iostream>
#include <algorithm>
#include <climits>

using namespace std;

pair<long long, long long> getMinMax(long long a[], int n) {
    long long ans1 = LLONG_MAX; // Initialize ans1 to the maximum possible long long value
    long long ans2 = LLONG_MIN; // Initialize ans2 to the minimum possible long long value

    for (int i = 0; i < n; i++) {
        ans1 = min(ans1, a[i]); // Update ans1 with the minimum value
        ans2 = max(ans2, a[i]); // Update ans2 with the maximum value
    }

    return make_pair(ans1, ans2);
}


// Don't write main function if you are coding in gfg
int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;

    long long a[n];
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    pair<long long, long long> result = getMinMax(a, n);

    cout << "Minimum: " << result.first << endl;
    cout << "Maximum: " << result.second << endl;

    return 0;
}
